src/m365/aad/commands/administrativeunit/administrativeunit-member-list.ts

```diff
@@ -0,0 +1,231 @@
+   */
+  properties?: string;
+
+  /** Filter expression used in $filter query parameter.*/
+  filter?: string;
+
+  /** If specified then $count=true is included.*/
+  count?: boolean;
+}
+
+class AadAdministrativeUnitMemberListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.ADMINISTRATIVEUNIT_MEMBER_LIST;
+  }
+
+  public get description(): string {
+    return 'Retrieves members (users, groups, or devices) of an administrative unit.';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'displayName'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        memberType: typeof args.options.memberType !== 'undefined',
+        properties: typeof args.options.properties !== 'undefined',
+        filter: typeof args.options.filter !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --administrativeUnitId [administrativeUnitId]'
+      },
+      {
+        option: '-n, --administrativeUnitName [administrativeUnitName]'
+      },
+      {
+        option: '-m, --memberType [memberType]',
+        autocomplete: ['user', 'group', 'device']
+      },
+      {
+        option: '-p, --properties [properties]'
+      },
+      {
+        option: '-f, --filter [filter]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.administrativeUnitId && !validation.isValidGuid(args.options.administrativeUnitId as string)) {
+          return `${args.options.administrativeUnitId} is not a valid GUID`;
+        }
+
+        if (args.options.memberType) {
+          if (['user', 'group', 'device'].indexOf(args.options.memberType) === -1) {
+            return `${args.options.memberType} is not a valid memberType value. Allowed values user|group|device`;
+          }
+        }
+
+        if (args.options.filter && !args.options.memberType) {
+          return 'Filter can be specified only if memberType is set';
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['administrativeUnitId', 'administrativeUnitName'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let administrativeUnitId = args.options.administrativeUnitId;
+
+    try {
+      if (args.options.administrativeUnitName) {
+        administrativeUnitId = (await aadAdministrativeUnit.getAdministrativeUnitByDisplayName(args.options.administrativeUnitName)).id;
+      }
+
+      const queryInputParameters = { properties: args.options.properties, filter: args.options.filter, count: false };
```

Comment:
Let's refactor this a bit. I don't think you need this intermediary object (and its interface)

Let's rename `createGraphQuery` to `getRequestUrl` and pass in `args.options` right here. Build the entire request URL in there instead of just the query params. 
You'll be able to do this in one place here and use the request URL below.

I think it will clean it up a bit like that.

Reply:
Changed